otx/cli/tools/build.py

```diff
@@ -103,11 +103,12 @@ def main():
                 raise ValueError("The selected backbone has inputs that the user must enter.")
             builder.merge_backbone(args.model, args.backbone)
 
-    config_manager.write_data_with_cfg(
-        workspace_dir=args.workspace_root,
-        train_data_roots=args.train_data_roots,
-        val_data_roots=args.val_data_roots,
-    )
+    if args.workspace_root is not None:
```

Comment:
If a user gives `--backbones` with `--workspace_root`, it still causes the problem.

So, I suggest moving this (line#107-111) into line#90. In this case we also could move line#81-82 to line#78

Reply:
Thanks for finding the problem with this code.
But your suggestion still generates `data.yaml` in unintended circumstances. 
In my opinion, supplementing the if statement in line 106 looks good.
How about `if not args.workspace_root and config_manager.splitted_dataset:` ?

Reply:
Oh, there is a problem with my suggestion. Let's think about it some more.

Reply:
Could you please check this https://github.com/openvinotoolkit/training_extensions/pull/1584/commits/c396f181c15821ee8c82fbb9daec5da9fa897bf3 ?

Reply:
It works as intended for the edge case you commented.

Reply:
Looks good :)